pkg/operator/ceph/cluster/cluster.go

```diff
@@ -63,6 +62,7 @@ type cluster struct {
 }
 
 func newCluster(c *cephv1.CephCluster, context *clusterd.Context) *cluster {
+	ownerRef := ClusterOwnerRef(c.Namespace, string(c.UID))
```

Comment:
Not an issue with this PR's changes, but something odd I see that might merit changing, the `ClusterOwnerRef` function accepts a string UID, and then it converts it again to a UID. Would it be better for this function to accept `type.UID`?

```go
func ClusterOwnerRef(namespace, clusterID string) metav1.OwnerReference {
	blockOwner := true
	return metav1.OwnerReference{
		APIVersion:         ClusterResource.Version,
		Kind:               ClusterResource.Kind,
		Name:               namespace,
		UID:                types.UID(clusterID),
		BlockOwnerDeletion: &blockOwner,
	}
}
```

Reply:
agreed, we should use the UID type instead